lib/SparqlView.js

```diff
@@ -52,25 +34,10 @@ class SparqlView extends ApiObject {
       })
     }
 
-    return Promise.resolve().then(() => {
-      if (this.source.value.endsWith('.ru')) {
-        return this.client.updateQuery(query, options)
-      } else {
-        return this.client.constructQuery(query, options)
-      }
-    }).then((result) => {
-      if (!result.ok) {
-        return Promise.reject(new Error(result.statusText))
-      }
-
-      return result.quadStream()
-    }).then((output) => {
-      if (output) {
-        res.graph(output)
-      } else {
-        res.status(204).end()
-      }
-    }).catch(next)
+    return Promise.resolve()
+      .then(() => this._executeQuery(query, options, req, res))
+      .then(next)
```

Comment:
On second thought I conclude that this should be explicitly called in the `SparqlQuery`/`SparqlUpdate` classes